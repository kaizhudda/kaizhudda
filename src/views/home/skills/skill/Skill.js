import React from 'react';
import SecondaryTitle from '../../../../sharedComponents/secondary-title'

import './Skill.scss';

const Skill = ({ skills, category }) => {
  return (
    <div className="Skills-content col-xs-12">
      <SecondaryTitle>{category}</SecondaryTitle>
      {
        skills.map((skill, index) => (
          <div key={index} className="skill">
            <div className="skill-title">{skill.name}</div>
            <div className="skill-score">{skill.experience}</div>
          </div>
        ))
      }

    </div>
  );
}

export default Skill; 
