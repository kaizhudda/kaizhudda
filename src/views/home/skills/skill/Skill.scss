@import '../../../../styles//variables';
@import '../../../../styles/media-queries';

.Skills-content {
  display: flex;
  flex-direction: column;
  width: 550px;
  margin-bottom: 25px;

  .skill {
    width: 100%;
    display: flex;
    justify-content: space-between;
    border-bottom: 1px solid rgba(45, 47, 47, 0.2);

    padding: 5px 0;
    &:hover {
      background: #efefef;
      
      .skill-score {
        font-size: 1.075rem;
        font-weight: bold;
      }
    }

    .skill-title {
      width: 70%;
      padding: 10px;
      text-align: left;
    }

    .skill-score {
      width: 30%;
      padding: 10px;
      text-align: right;
      color: #22a39f;
      transition: all 0.2s cubic-bezier(.25,.8,.25,1);
    }
  }
}