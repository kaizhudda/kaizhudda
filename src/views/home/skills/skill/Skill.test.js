import React from 'react';
import { shallow } from 'enzyme';
import Skill from './Skill';

const skills = [
  {
    name: "React",
    experience: "3+ years"
  },
  {
    name: "Redux",
    experience: "3+ years"
  },
];
let props = {
  skills,
  category: 'Test'
}
let wrapped = shallow(<Skill {...props} />); 

describe('Skill', () => {
  it('should render the Skill Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('should render 2 skills', () => {
    expect(wrapped.find('.skill').length).toBe(2);
  });

  it('should render Secondary Title', () => {
    expect(wrapped.find('SecondaryTitle').exists()).toBe(true);
  });
});
