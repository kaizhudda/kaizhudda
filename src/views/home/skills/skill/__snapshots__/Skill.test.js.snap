// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skill should render the Skill Component correctly 1`] = `
<div
  className="Skills-content col-xs-12"
>
  <SecondaryTitle>
    Test
  </SecondaryTitle>
  <div
    className="skill"
    key="0"
  >
    <div
      className="skill-title"
    >
      React
    </div>
    <div
      className="skill-score"
    >
      3+ years
    </div>
  </div>
  <div
    className="skill"
    key="1"
  >
    <div
      className="skill-title"
    >
      Redux
    </div>
    <div
      className="skill-score"
    >
      3+ years
    </div>
  </div>
</div>
`;
